Pollo/ViewControllers/CardController+Extension.swift

```diff
@@ -220,27 +221,41 @@ extension CardController: UIScrollViewDelegate {
 
 extension CardController: SocketDelegate {
 
-    func sessionConnected() {}
+    func sessionConnected() {
+        let banner = NotificationBanner.connectedBanner()
+        BannerController.shared.show(banner)
+        collectionView.isUserInteractionEnabled = true
+    }
     
-    func sessionDisconnected() {}
-
-    func sessionErrored() {
+    func sessionDisconnected() {
         socket.socket.connect(timeoutAfter: 5) { [weak self] in
             guard let `self` = self else { return }
-            let alertController = self.createAlert(title: "Error", message: "Could not join poll. Try joining again!", handler: { _ in
-                guard let viewControllers = self.navigationController?.viewControllers else { return }
-                if viewControllers.count > 3 {
-                    // Pop back to PollsViewController
-                    guard let viewController = viewControllers[viewControllers.count - 3] as? PollsViewController else { return }
-                    self.navigationController?.popToViewController(viewController, animated: true)
-                    self.socket.socket.disconnect()
-                    self.socket.delegate = nil
-                    viewController.pollsDateViewControllerWasPopped(for: self.userRole)
-                }
-            })
-            if self.presentedViewController == nil {
-                self.present(alertController, animated: true, completion: nil)
+            let banner = NotificationBanner.disconnectedBanner()
+            banner.onTap = { [weak self] in
+                guard let `self` = self else { return }
+                self.socket.socket.setReconnecting(reason: "")
```

Comment:
When there's no reason provided the banner just says "Trying to reconnect..." since `sessionDisconnected()` isn't called with a specific reason (which could be multiple, i.e. `The request timed out` and `No Internet connection`). Basically when the user taps the disconnected banner i think the reason that it disconnected in the first place is irrelevant.